python/test/unit/language/test_conversions.py

```diff
@@ -326,13 +326,17 @@ def test_typeconvert_downcast(src_dtype, dst_dtype, rounding, max_repr, device):
     if src_dtype != 'float32' and is_cuda() and torch.cuda.get_device_capability(0) < (9, 0):
         pytest.skip("non-float32 downcast tests only supported on NVGPU with compute capability 9.0+")
 
-    if dst_dtype in ('float8e5', 'float8e4nv') and rounding == 'rtne' and (is_hip() or torch.cuda.get_device_capability(0) < (9, 0)):
+    is_fp16_to_fp8e4nv_on_hip = is_hip() and (src_dtype == 'float16') and \
+        (dst_dtype == 'float8e4nv') and (rounding == 'rtne')
+
+    if not is_fp16_to_fp8e4nv_on_hip and dst_dtype in ('float8e5', 'float8e4nv') \
```

Comment:
Nit: it reads better if we put `and not is_fp16_to_fp8e4nv_on_hip` as the last piece. Similarly for L339.